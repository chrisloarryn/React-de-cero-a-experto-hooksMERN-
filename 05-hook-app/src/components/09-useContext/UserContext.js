import { createContext } from 'react'

export const UserContext = createContext({
  user: {},
})

// export const UserProvider = ({ children }) => {
//   return (
//     <UserContext.Provider value={{ user: {} }}>{children}</UserContext.Provider>
//   )
// }
