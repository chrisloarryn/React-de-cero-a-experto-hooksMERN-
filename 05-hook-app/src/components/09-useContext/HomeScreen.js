import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {
  // ref to context
  const userContext = useContext(UserContext)
  const { user } = userContext
  return (
    <div>
      <h1>HomeScreen {`${user?.id || 'PLEASE'} ${user?.name || 'LOGIN'}`}</h1>
      <hr />

      <pre>{JSON.stringify(user, null, 3)}</pre>
    </div>
  )
}
