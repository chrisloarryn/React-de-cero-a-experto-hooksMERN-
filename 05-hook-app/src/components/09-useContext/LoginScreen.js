import React, { useContext, useEffect } from 'react'
import { useHistory } from 'react-router-dom'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext)
  const history = useHistory()
  useEffect(() => {
    user && user.id && history.push('/')
  }, [user, history])
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() =>
          setUser({
            id: 234,
            name: 'namdsssde'
          })
        }>
        Login
      </button>
    </div>
  )
}
