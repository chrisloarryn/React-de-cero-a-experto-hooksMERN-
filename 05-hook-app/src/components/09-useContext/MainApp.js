import React, { useState, useEffect } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {
  const [user, setUser] = useState({})
  useEffect(() => {
    console.log('user has been changed!')
  }, [user])
  return (
    <UserContext.Provider
      value={{
        user,
        setUser,
      }}>
      <AppRouter />
    </UserContext.Provider>
  )
}
