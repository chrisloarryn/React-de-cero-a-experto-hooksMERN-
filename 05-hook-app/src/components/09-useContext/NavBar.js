import React, { useContext } from 'react'
import { Link, NavLink } from 'react-router-dom'
import { UserContext } from './UserContext'

export const NavBar = () => {
  const { user } = useContext(UserContext)

  return (
    <>
      <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <Link className="navbar-brand" to="/">
          useContext
        </Link>

        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink exact activeClassName="active" className="nav-link" to="/">
              Home
            </NavLink>
            <NavLink
              exact
              activeClassName="active"
              className="nav-link"
              to="/about">
              About
            </NavLink>
            {!user.id && (
              <NavLink
                exact
                activeClassName="active"
                className="nav-link"
                to="/login">
                Login
              </NavLink>
            )}
          </div>
        </div>
      </nav>
      {/* <nav>
    <ul>
      <li>
        <Link to="./">Home</Link>
      </li>
      <li>
        <Link to="./about">About</Link>
      </li>
      <li>
        <Link to="./login">Login</Link>
      </li>
    </ul>
  </nav> */}
    </>
  )
}
