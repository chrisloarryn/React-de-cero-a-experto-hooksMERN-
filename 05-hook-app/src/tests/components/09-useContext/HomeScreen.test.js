import React from 'react'
import { mount } from 'enzyme'
import { HomeScreen } from 'components/09-useContext/HomeScreen'
import { UserContext } from 'components/09-useContext/UserContext'
import '@testing-library/jest-dom'

describe('Tests in <HomeScreen /> ', () => {
  const user = {
    name: 'Christopher',
    email: 'christopher@gmail.com'
  }
  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  )

  test('Should be rendered correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
