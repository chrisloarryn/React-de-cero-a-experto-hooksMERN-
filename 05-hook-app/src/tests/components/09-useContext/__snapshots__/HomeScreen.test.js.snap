// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <HomeScreen />  Should be rendered correctly 1`] = `
<div>
  <h1>
    HomeScreen 
    PLEASE Christopher
  </h1>
  <hr />
  <pre>
    {
   "name": "Christopher",
   "email": "christopher@gmail.com"
}
  </pre>
</div>
`;
